src/main/java/persistence/entity/EntitySnapshot.java

```diff
@@ -0,0 +1,49 @@
+package persistence.entity;
+
+import persistence.util.ReflectionUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class EntitySnapshot {
+    private Object originalEntity;
+    private Map<String, Object> keyValues = new HashMap<>();
+    private boolean isDirty;
+
+    public EntitySnapshot(Object entity) {
+        this.originalEntity = entity;
+        this.keyValues = ReflectionUtil.getAllFieldNameAndValue(entity);
+        this.isDirty = false;
+    }
```

Comment:
```suggestion
public class EntitySnapshot {
    private Object originalEntity;
    private Map<String, Object> keyValues;
    private boolean isDirty;

    public EntitySnapshot(Object entity) {
        this.originalEntity = entity;
        this.keyValues = ReflectionUtil.getAllFieldNameAndValue(entity);
        this.isDirty = false;
    }
```

사소하지만 항상 새로 초기화 되어서 미리 필드를 초기화 해놓지 않아도 될 것 같아요 👍